### ISSUE TITLE
warn user if client version != server version

### ISSUE THREAD

**Issue by tim-o** [cockroachdb/cockroach#28507]
**FEATURE REQUEST**

Running inconsistent versions (for instance, connecting to a 2.x cluster with a 1.x client) can result in inconsistent behavior and commands failing (ex: http://forum.cockroachlabs.com/t/1842)

It'd be good to take the extra step and warn users if the client version != server version.

**Comment by tim-o** [comment]
Zendesk ticket #[2717](https://cockroachdb.zendesk.com/agent/tickets/2717) has been linked to this issue.

**Comment by knz** [comment]
@tim-o we have addressed this in the 2.0 client already, which print the difference:

```
# Client version: CockroachDB CCL v2.0.4-24-g3f2f0f5ee (x86_64-unknown-freebsd12.0, built 2018/08/15 10:48:43, go1.10.3)
# Server version: CockroachDB CCL v2.1.0-alpha.20180702-1701-gffb539f645-dirty (x86_64-unknown-freebsd12.0, built 2018/08/17 07:44:17, go1.10.3)
```

The printing of the two versions was added in #16703.

We do also print a warning when the difference is in the other direction:

```
# Client version: CockroachDB CCL v2.1.0-alpha.20180702-1701-gffb539f645-dirty (x86_64-unknown-freebsd12.0, built 2018/08/17 07:44:17, go1.10.3)
# Server version: CockroachDB CCL v2.0.4-24-g3f2f0f5ee (x86_64-unknown-freebsd12.0, built 2018/08/15 10:48:43, go1.10.3)

warning: server version older than client! proceed with caution; some features may not be available.
```

This warning was added in #20935; it is also present in 1.1 thanks to #20945.

We do not print the opposite warning (server newer than client) because as @bdarnell points out in https://github.com/cockroachdb/cockroach/issues/21323#issuecomment-356114420 clients should be able to connect to newer servers.

Except that, as you found out, the 1.x client doesn't work well with 2.x servers.

Now we could add that check to 1.1 but 1.1 is EOL and we don't do feature change on it any more. @bdarnell correct me if I am mistaken.
